gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -187,28 +190,33 @@ impl ClnRpcService {
             ClnExtensionError::Error(anyhow!(e))
         })
     }
+
+    pub async fn pubkey(&self) -> Result<PublicKey, ClnExtensionError> {
+        self.rpc_client()
+            .await?
+            .call(cln_rpc::Request::Getinfo(
+                model::requests::GetinfoRequest {},
+            ))
+            .await
+            .map(|response| match response {
+                cln_rpc::Response::Getinfo(model::GetinfoResponse { id, .. }) => Ok(id),
+                _ => Err(ClnExtensionError::Error(anyhow!(
+                    "Unexpected response from cln_rpc"
```

Comment:
I would still go for `WrongResponse` [as pointed out in this comment](https://github.com/fedimint/fedimint/pull/1671#discussion_r1105928951).

edit: now that I read the comment again it might be a bit ambiguous. I'm in favor of `WrongReponse` because that immediately points out that we got a wrong response. `Unexpected response` isn't that great, imho, because we *did* expect a response, so it was not an "unexpected response".